frontend/src/containers/main/Main.tsx

```diff
@@ -45,6 +48,18 @@ function Main() {
     });
   }, []);
 
+  useEffect(() => {
+    SetCurrentUser(APIresult?.currentUser);
+    if (APIresult?.currentUser) {
+      localStorage.setItem('loggedIn', 'true');
+    }
+  }, [APIresult])
+
+  useEffect(() => {
+    if (APIerror) {
+      localStorage.removeItem('loggedIn');
+    }
+  }, [APIerror])
```

Comment:
Using localStorage to have a smoother auth checking experience for the client side. localStorage will process faster than API responses when the user is already signed in, thus having a more consistent and smooth experience when rendering the profile page.